src/04-01.zig

```diff
@@ -12,8 +12,8 @@ pub fn main() !void {
     defer _ = gpa.deinit();
     const allocator = gpa.allocator();
 
-    const loopback = try net.Ip4Address.parse("127.0.0.1", 0);
-    const localhost = net.Address{ .in = loopback };
+    const loopback = try net.Ip6Address.parse("::1", 0);
```

Comment:
This is a good idea. 👍

How about pass an argument here to represent which protocol to use?

Reply:
The problem is that it will complicate the code, while the recipes in the cookbook are intended, I believe, to be simple and self-sufficient. I just added in the PR `IP-VERSION-04-01.zig` so you can get an idea of the Zig concepts it will depend on. (And it will be worse if the IP version is not comptime-known.)

Reply:
Thanks for clarification.

In order to keep code simple, maybe we only introduce ipv6 in docs, not code.
- https://github.com/zigcc/zig-cookbook/blob/main/book-src/04-01-tcp-server.md

Do you like this idea?

Reply:
I would prefer IPv6 to be the default (and IPv4 explained in the docs). After all, it is no longer the 20th century.

Reply:
I don't think this is a good idea, IPv4 still has more adoption than IPv6. And the cookbook should prefer command task that everyone may use everyday.

- https://thenewstack.io/why-is-ipv6-adoption-slow/